python/ray/autoscaler/_private/commands.py

```diff
@@ -691,6 +693,12 @@ def get_or_create_head_node(config: Dict[str, Any],
             setup_commands = config["head_setup_commands"]
             ray_start_commands = config["head_start_ray_commands"]
 
+        # If restarting Ray on a manually-managed on-prem cluster,
+        # we need to sync local and head representations of cluster state.
+        if ray_start_commands and is_local_manual(config["provider"]):
+            # Add cluster state file to file mounts.
+            config = sync_state(config)
```

Comment:
not sure why ray_start_commands here are relevant.

Reply:
This isn't well-documented here and I'll add a comment explaining this:
The presence of `ray_start_commands` here indicates that you're restarting the cluster.
If you're not restarting the cluster, then doing the sync will break the autoscaler's state because the machine running ray up doesn't know about the worker nodes.